tests/aws/conftest.py

```diff
@@ -78,110 +47,5 @@ def pytest_runtestloop(session):
-
-
-def startup_monitor() -> None:
-    """
-    The startup monitor is a thread that waits for the startup_monitor_event and, once the event is true, starts a
-    localstack instance in its own thread context.
-    """
-    logger.info("waiting on localstack_start signal")
-    startup_monitor_event.wait()
-
-    if localstack_stop.is_set():
-        # this is called if _trigger_stop() is called before any test has requested the localstack_runtime fixture.
-        logger.info("ending startup_monitor")
-        localstack_stopped.set()
-        return
-
-    if is_env_true("TEST_SKIP_LOCALSTACK_START") or os.environ.get("TEST_TARGET") == "AWS_CLOUD":
-        logger.info("TEST_SKIP_LOCALSTACK_START is set, not starting localstack")
-        localstack_started.set()
-        localstack_stopped.set()
-        return
-
-    logger.info("running localstack")
-    run_localstack()
-
-
-def run_localstack():
-    """
-    Start localstack and block until it terminates. Terminate localstack by calling _trigger_stop().
-    """
-    # configure
-    os.environ[ENV_INTERNAL_TEST_RUN] = "1"
-    safe_requests.verify_ssl = False
-    localstack_config.FORCE_SHUTDOWN = False
-    localstack_config.GATEWAY_LISTEN = [
-        localstack_config.HostAndPort(host="0.0.0.0", port=constants.DEFAULT_PORT_EDGE)
-    ]
-
-    def watchdog():
-        logger.info("waiting stop event")
-        localstack_stop.wait()  # triggered by _trigger_stop()
-        logger.info("stopping infra")
-        infra.stop_infra()
-
-    monitor = threading.Thread(target=watchdog)
-    monitor.start()
-
-    logger.info("starting localstack infrastructure")
-    infra.start_infra(asynchronous=True)
-
     for fn in test_init_functions:
-        try:
-            # asynchronous init functions
-            fn()
-        except Exception:
-            logger.exception("exception while running init function for test")
-
-    logger.info("waiting for infra to be ready")
-    events.infra_ready.wait()  # wait for infra to start (threading event)
-    localstack_started.set()  # set conftest inter-process Event
-
-    logger.info("waiting for shutdown")
-    try:
-        logger.info("waiting for watchdog to join")
-        monitor.join()
-    finally:
-        logger.info("ok bye")
-        localstack_stopped.set()
-
-
-@pytest.fixture(scope="session", autouse=True)
-def localstack_runtime():
-    """
-    This is a dummy fixture. Each test requests the fixture, but it actually just makes sure that localstack is running,
-    blocks until localstack is running, or starts localstack the first time the fixture is requested.
-    It doesn't actually do anything but signal to the `startup_monitor` function.
-    """
-    if localstack_started.is_set():
-        # called by all tests after the startup has completed and the initial tests are unblocked
-        yield
-        return
-
-    startup_monitor_event.set()
-    localstack_started.wait()
-    yield
-    return
+        fn()
```

Comment:
Maybe we should not have stripped the try: except: here, but if it cannot install the test dependencies, it might be good to fail immediately.